### ISSUE TITLE
TQDM progress bar broken

### ISSUE THREAD

**Issue by devxpy** [tensorflow/addons#812]
**System information**
- Google Colab
- tfa-nightly 0.7.0.dev20191219
- tqdm 4.41.0

The progress bar just skips to end at the beginning of each epoch and stays there till the epoch completes. The metrics don't update either.

I am using `fit_generator`.

![image](https://user-images.githubusercontent.com/19492893/71453401-14360280-27b1-11ea-8109-473b7d934ae9.png)

**Comment by seanpmorgan** [comment]
Hi @devxpy thanks for the report. Could you please provide a minimal reproducible example.

CC @shun-lin code owner.

**Comment by shun-lin** [comment]
Hi @devxpy, have you turn off the default progress bar of Keras by passing in the keyword `verbose=0` to `model.fit`?

**Comment by RomainBrault** [comment]
This is due to using generator with steps_per_epoch=None (default).

Then self.params['steps'] is None and TQDMProgressBar breaks.

**Comment by shun-lin** [comment]
got it, I was able to reproduce the issue:

I found the bug and it was when we are training using `generator` (with `fit_generator` or `fit` with generator). It arises from my misunderstanding of how the original progress bar updates itself with generator. At the end of each `batch` the progress bar is updated by 1 instead of `batch_size`, and therefore `TQDMProgressBar` is progressing (incorrectly) `batch_size` times faster. The change is pretty simple (~2 lines change) deleting the `if` statement in the beginning of `on_epochs_end` as both `fit` and `fit_generator` will update the progress bar in the same way. I will submit a PR later today. Thanks @RomainBrault
